python/ray/dashboard/agent.py

```diff
@@ -47,6 +48,9 @@ def __init__(
         # Public attributes are accessible for all agent modules.
         self.ip = node_ip_address
         self.minimal = minimal
+        self.thread_pool_executor = ThreadPoolExecutor(
```

Comment:
Please check my comment above.

We need to
 - Isolate these TPEs to non critical operations (like refreshing stats/data used in UI)
 - Limit concurrency of these TPEs (to 2-4)